### PR TITLE
Fix swapchain acquire count check

### PR DESCRIPTION
**Pull request by krOoze**
- add proper VUIDs
- use `timeout`
- make error message more friendly
- add tests

### CODE REVIEW

**Comment by mark-lunarg** [comment]
Running through CI.

**Comment by mark-lunarg** [comment]
@krOoze, this PR fails `VkPositiveLayerTest.SwapChainImageLayout` on the Android platforms (including my Pixel2XL).

**Comment by mark-lunarg** [comment]
For this one, the real failures are `VkLayerTest.SwapchainAcquireTooManyImages2KHR` and `VkLayerTest.WarningSwapchainCreateInfoPreTransform`:


```
2788 11-14 15:31:17.670  1207  1279 I VulkanLayerValidationTests: [ RUN      ] VkLayerTest.SwapchainAcquireTooManyImages
2789 11-14 15:31:17.681  1207  1279 I vulkan  : Loaded layer VK_LAYER_KHRONOS_validation
2790 11-14 15:31:17.684  1207  1279 I Adreno  : QUALCOMM build          : 6fb5a5b, Ife855c4895
2791 11-14 15:31:17.684  1207  1279 I Adreno  : Build Date              : 08/21/18
2792 11-14 15:31:17.684  1207  1279 I Adreno  : Shader Compiler Version : EV031.25.00.00
2793 11-14 15:31:17.684  1207  1279 I Adreno  : Local Branch            : Googledrop_0815
2794 11-14 15:31:17.684  1207  1279 I Adreno  : Remote Branch           :
2795 11-14 15:31:17.684  1207  1279 I Adreno  : Remote Branch           :
2796 11-14 15:31:17.684  1207  1279 I Adreno  : Reconstruct Branch      :
2797 11-14 15:31:17.684  1207  1279 I Adreno  : Build Config                     : S L 4.0.10 AArch64
2798 11-14 15:31:17.685  1207  1279 I ConfigStore: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasHDRDisplay retrieved: 0
2799 11-14 15:31:17.685  1207  1279 I chatty  : uid=10310(com.example.VulkanLayerValidationTests) identical 4 lines
2800 11-14 15:31:17.685  1207  1279 I ConfigStore: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasHDRDisplay retrieved: 0
2801 11-14 15:31:17.695   613   684 I SDM     : HWCDisplay::GetColorModeCount: Supported color mode count = 3
2802 11-14 15:31:17.695   613   684 I SDM     : HWCDisplay::GetColorModes: Supports color mode[0] = 0
2803 11-14 15:31:17.695   613   684 I SDM     : HWCDisplay::GetColorModes: Supports color mode[1] = 7
2804 11-14 15:31:17.695   613   684 I SDM     : HWCDisplay::GetColorModes: Supports color mode[2] = 9
2805 11-14 15:31:17.696  1207  1279 I ConfigStore: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasWideColorDisplay retrieved: 1
2806 11-14 15:31:17.696   613   684 I SDM     : HWCDisplay::GetColorModeCount: Supported color mode count = 3
2807 11-14 15:31:17.697   613   684 I SDM     : HWCDisplay::GetColorModes: Supports color mode[0] = 0
2808 11-14 15:31:17.697   613   684 I SDM     : HWCDisplay::GetColorModes: Supports color mode[1] = 7
2809 11-14 15:31:17.697   613   684 I SDM     : HWCDisplay::GetColorModes: Supports color mode[2] = 9
2810 11-14 15:31:17.697  1207  1279 I ConfigStore: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasWideColorDisplay retrieved: 1
2811 11-14 15:31:17.712  1207  1279 I VulkanLayerValidationTests: [       OK ]VkLayerTest.SwapchainAcquireTooManyImages (42 ms)
2812 11-14 15:31:17.712  1207  1279 I VulkanLayerValidationTests: [ RUN      ] VkLayerTest.SwapchainAcquireTooManyImages2KHR
2813 11-14 15:31:17.723  1207  1279 I vulkan  : Loaded layer VK_LAYER_KHRONOS_validation
2814 11-14 15:31:17.726  1207  1279 I Adreno  : QUALCOMM build          : 6fb5a5b, Ife855c4895
2815 11-14 15:31:17.726  1207  1279 I Adreno  : Build Date              : 08/21/18
2816 11-14 15:31:17.726  1207  1279 I Adreno  : Shader Compiler Version : EV031.25.00.00
2817 11-14 15:31:17.726  1207  1279 I Adreno  : Local Branch            : Googledrop_0815
2818 11-14 15:31:17.726  1207  1279 I Adreno  : Remote Branch           :
2819 11-14 15:31:17.726  1207  1279 I Adreno  : Remote Branch           :
2820 11-14 15:31:17.726  1207  1279 I Adreno  : Reconstruct Branch      :
2821 11-14 15:31:17.726  1207  1279 I Adreno  : Build Config                     : S L 4.0.10 AArch64
2822 11-14 15:31:17.727  1207  1279 I ConfigStore: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasHDRDisplay retrieved: 0
2823 11-14 15:31:17.727  1207  1279 I chatty  : uid=10310(com.example.VulkanLayerValidationTests) identical 8 lines
2824 11-14 15:31:17.727  1207  1279 I ConfigStore: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasHDRDisplay retrieved: 0
2825 11-14 15:31:17.737   611  1284 E BufferQueueProducer: [com.example.VulkanLayerValidationTests/android.app.NativeActivity#0] connect: already connected (cur=1 req=1)
2826 11-14 15:31:17.737  1207  1279 E vulkan  : native_window_api_connect() failed: Invalid argument (-22)
2827 11-14 15:31:17.737  1207  1279 I VulkanLayerValidationTests: *** Failure in jni/../../tests/vklayertests_others.cpp:1250 Value of: InitSwapchain()
2828 11-14 15:31:17.737  1207  1279 I VulkanLayerValidationTests:   Actual: false
2829 11-14 15:31:17.737  1207  1279 I VulkanLayerValidationTests: Expected: true
2830 11-14 15:31:17.744  1207  1279 I VulkanLayerValidationTests: [  FAILED  ]VkLayerTest.SwapchainAcquireTooManyImages2KHR (32 ms)
```


```
3363 11-14 15:31:19.286  1207  1279 I VulkanLayerValidationTests: [ RUN      ] VkLayerTest.WarningSwapchainCreateInfoPreTransform
3364 11-14 15:31:19.299  1207  1279 I vulkan  : Loaded layer VK_LAYER_KHRONOS_validation
3365 11-14 15:31:19.302  1207  1279 I Adreno  : QUALCOMM build          : 6fb5a5b, Ife855c4895
3366 11-14 15:31:19.302  1207  1279 I Adreno  : Build Date              : 08/21/18
3367 11-14 15:31:19.302  1207  1279 I Adreno  : Shader Compiler Version : EV031.25.00.00
3368 11-14 15:31:19.302  1207  1279 I Adreno  : Local Branch            : Googledrop_0815
3369 11-14 15:31:19.302  1207  1279 I Adreno  : Remote Branch           :
3370 11-14 15:31:19.302  1207  1279 I Adreno  : Remote Branch           :
3371 11-14 15:31:19.302  1207  1279 I Adreno  : Reconstruct Branch      :
3372 11-14 15:31:19.302  1207  1279 I Adreno  : Build Config                     : S L 4.0.10 AArch64
3373 11-14 15:31:19.303  1207  1279 I ConfigStore: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasHDRDisplay retrieved: 0
3374 11-14 15:31:19.303  1207  1279 I chatty  : uid=10310(com.example.VulkanLayerValidationTests) identical 4 lines
3375 11-14 15:31:19.303  1207  1279 I ConfigStore: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasHDRDisplay retrieved: 0
3376 11-14 15:31:19.314   611  1284 E BufferQueueProducer: [com.example.VulkanLayerValidationTests/android.app.NativeActivity#0] connect: already connected (cur=1 req=1)
3377 11-14 15:31:19.314  1207  1279 E vulkan  : native_window_api_connect() failed: Invalid argument (-22)
3378 11-14 15:31:19.314  1207  1279 I VulkanLayerValidationTests: *** Failure in jni/../../tests/vkrenderframework.cpp:145 Failed
3379 11-14 15:31:19.314  1207  1279 I VulkanLayerValidationTests: Did not receive expected error 'UNASSIGNED-CoreValidation-SwapchainPreTransform'
3380 11-14 15:31:19.321  1207  1279 I VulkanLayerValidationTests: [  FAILED  ]VkLayerTest.WarningSwapchainCreateInfoPreTransform (35 ms)
```

**Comment by krOoze** [comment]
@mark-lunarg Probably same problem that afflicts #1374. I pushed the same workaround attempt.

**Comment by mark-lunarg** [comment]
Ran through CI, rebased, squashed, and pushed to Master.  What seemed to make a difference was the DestroySwapchain calls you added.  Android is different from desktop in that a single instance of the 'window' object is created for all the tests, and without the destroy calls some state appeared to persist.  I'm looking into why we didn't get any messages for not calling DestroySwapchain to begin with, or why the framework didn't clean up after itself properly.

File: tests/vklayertests_others.cpp [hunk 1]
```diff
@@ -1254,6 +1254,113 @@ TEST_F(VkLayerTest, SwapchainAcquireImageNoSync2KHR) {
...
+
+    ASSERT_NO_FATAL_FAILURE(InitState());
+    ASSERT_TRUE(InitSwapchain());
+    uint32_t image_count;
+    ASSERT_VK_SUCCESS(vk::GetSwapchainImagesKHR(device(), m_swapchain, &image_count, nullptr));
+    VkSurfaceCapabilitiesKHR caps;
+    ASSERT_VK_SUCCESS(vk::GetPhysicalDeviceSurfaceCapabilitiesKHR(gpu(), m_surface, &caps));
+
+    const uint32_t acquirable_count = image_count - caps.minImageCount + 1;
+    uint32_t image_i = 0;  // MockICD does not write this so trick it to return sane value
```
Comment (krOoze) [inline tests/vklayertests_others.cpp]: In retrospect I am bit surprised this works (should increment the `image_i`). Meanwhile I have trouble setting Mock locally; it just says the Surface extension is not supported and skips the whole test.

**Pull request closed without merging**